### ISSUE TITLE
🐛 BUG: Inconsistent rendering of code blocks in Markdown

### ISSUE THREAD

**Issue by madx** [snowpackjs/astro#699]
### What package manager are you using?

npm

### What operating system are you using?

Linux

### Describe the Bug

Code blocks in Markdown are inconsistently syntax highlighted depending on how you render your Markdown.

- When using the Markdown component with inline Markdown (passed as children), code is properly highlighted
- When using the Markdown component with the `content` prop code is not highlighted
- When using `Astro.fetchContent` + `item.astro.html`code is not highlighted
- When using `.md` pages, code is properly highlighted

### Steps to Reproduce

You can see the attached minimal reproducible example for a demo of those cases

```
git clone https://github.com/madx/astro-inconsistent-code-render-markdown.git
cd astro-inconsistent-code-render-markdown
npm i
npm start
open http://127.0.0.1:3000/
```

### Link to Minimal Reproducible Example (Optional)

https://github.com/madx/astro-inconsistent-code-render-markdown

**Comment by madx** [comment]
I took a look around in the code and I think Prism is not called in `renderMarkdown` (from `@astro/markdown-support`.

I did not really undersand how the markdown is parsed in the case of the contents being inserted in the `<slot />` of the `Markdown` component

**Comment by FredKSchott** [comment]
@matthewp would be good to take a look at this if you can

**Comment by eallenOP** [comment]
+1 from me. I can't get syntax highlighting inside the <Markdown> block in a normal .astro file which also mixes in HTML.

Example: `src/pages/about.astro`

```astro
---
import { Markdown } from 'astro/components';
import BaseLayout from '../layouts/BaseLayout.astro';
---

<BaseLayout>
    <main>
        <h1>Astro Example</h1>
        <article>
            <Markdown>
            ## Set up an Astro project

            First create a new directory for your project

            ```shell
            ~ $ mkdir my-project
            ~ $ cd my-project
            my-project $
            ```
            
            More paragraphs and so forth...
            </Markdown>
        </article>
    </main>
</BaseLayout>
```

When the above is rendered (and other code blocks with other languages specified on this same page) I do not get Prism syntax highlighting. Am I supposed to switch it on somewhere? Or is this already being fixed?

(It sounds from the OP here that some people *are* getting proper highlighting in this situation...)

**Comment by madx** [comment]
I updated my reproduction example repository to Astro 0.18.12 to handle changes that happened since. The Markdown rendering still show the same inconsistencies (which is what I expected since this issue has not been resolved yet). 

@eallenOP In my reproduction example I have the following test case in which Prism highlighting is properly applied:

```js
<div>
  <h1>Inline Markdown in <code>&lt;Markdown&gt;</code> component</h1>
  <Markdown>
  ```js
  console.log("Hello, world")
  ```
  </Markdown>
</div>
```

Any ETA on when this will be resolved? This is currently preventing me for using Astro for my personal blog since I can't build the post list index which is using `Astro.fetchContent` (not a big deal, I can wait lol)

**Comment by pyronaur** [comment]
I ran into this today as well - this makes Astro unusable for me at the moment, until I can find at least a workaround. All my blog posts that involve code are affected by this, so I'm unable to use Astro on my live site at the moment.

**Comment by FredKSchott** [comment]
@natemoo-re has been doing a ton of work to get Markdown working in the next release of Astro. We'll make sure that any remaining bugs are tackled in that release, and then would love all of your help testing to confirm

**Comment by pyronaur** [comment]
My current workaround is a `MarkdownFix.astro` component:

```
---
import hljs from 'highlight.js';
import markdownit from 'markdown-it';

const markdown = new markdownit({
  highlight: function (str, lang) {
    if (lang && hljs.getLanguage(lang)) {
      try {
        return '<pre class="hljs"><code>' +
               hljs.highlight(str, { language: lang, ignoreIllegals: true }).value +
               '</code></pre>';
      } catch (__) {}
    }

    return '<pre class="hljs"><code>' + markdown.utils.escapeHtml(str) + '</code></pre>';
  }
});

const {content} = Astro.props
---

{markdown.render(content)}
```

This seems to do the trick. Excited to see Astro.next though  🙂

**Comment by FredKSchott** [comment]
Thanks for the report! We believe that this has been fixed in astro v0.21, going out today. Let us know if you can still reproduce the issue after updating to v0.21 and we'll be happy to take a look!
